### PR TITLE
Added Compose Previews for NFC, Onboarding & Companion Views

### PR DESCRIPTION
**Pull request by vieck**
## Summary
Adding in more compose previews for new (and experienced) devs to remember what the screens look like at a glance. Allows for easier font sizing, style checking, etc

## Screenshots
ChooseEntityView
DiscoveryView
ManualSetupView
NFCEditView
NFCWelcomeView
NFCWriteView Disabled
NFCWriteView Enabled
TagReaderView

## Link to pull request in Documentation repository

Documentation: home-assistant/companion.home-assistant#

### CODE REVIEW

File: wear/src/main/java/io/homeassistant/companion/android/views/ChooseEntityView.kt [hunk 1]
```diff
@@ -150,3 +155,46 @@ private fun ChooseEntityChip(
         colors = getFilledTonalButtonColors()
     )
 }
+
+@Preview(showSystemUi = true)
```
Comment (dshokouhi) [inline wear/src/main/java/io/homeassistant/companion/android/views/ChooseEntityView.kt:159]: Just to match the other Wear device screens can we set the device to large round? The screenshot in the description also shows a wearable screen against a phone UI :)

Comment (vieck) [reply wear/src/main/java/io/homeassistant/companion/android/views/ChooseEntityView.kt]: Done!

File: wear/src/main/java/io/homeassistant/companion/android/views/ChooseEntityView.kt [hunk 2]
```diff
@@ -150,3 +156,46 @@ private fun ChooseEntityChip(
...
+fun ChooseEntityView_WithData_Preview() {
+    ChooseEntityView(
+        entitiesByDomainOrder = remember {
+            mutableStateListOf("first", "second")
+        },
+        entitiesByDomain = remember {
+            mutableStateMapOf(
+                Pair(
+                    "first",
+                    mutableStateListOf(Entity("first", "on", "", Calendar.getInstance(), Calendar.getInstance(), null))
```
Comment (dshokouhi) [inline wear/src/main/java/io/homeassistant/companion/android/views/ChooseEntityView.kt:188]: Sorry just noticed you had some mock entities, we actually have some you can use for this purpose :)

https://github.com/home-assistant/android/blob/master/wear/src/main/java/io/homeassistant/companion/android/util/PreviewData.kt#L34

Comment (vieck) [reply wear/src/main/java/io/homeassistant/companion/android/views/ChooseEntityView.kt]: Done

File: app/src/main/java/io/homeassistant/companion/android/onboarding/manual/ManualSetupView.kt [hunk 3]
```diff
@@ -79,3 +94,17 @@ fun ManualSetupView(
         }
     }
 }
+
+@Preview(showSystemUi = true)
```
Comment (dshokouhi) [inline app/src/main/java/io/homeassistant/companion/android/onboarding/manual/ManualSetupView.kt:98]: is this first annotation necessary here?

Comment (vieck) [reply app/src/main/java/io/homeassistant/companion/android/onboarding/manual/ManualSetupView.kt:98]: @dshokouhi Yes, it shows one for dark mode and one for standard mode. Mostly just to compare the two at a glance when developing

**Review by jpelgrom — COMMENTED** [review]
Functionally this looks good! A couple of small suggestions to clean up the code and for consistency.

(Also, unless there is a merge conflict or requested, you don't need to rebase/merge the latest commits to the branch.)

File: app/src/main/java/io/homeassistant/companion/android/onboarding/discovery/DiscoveryView.kt [hunk 4]
```diff
@@ -18,34 +18,51 @@ import androidx.compose.material.Text
...
+import io.homeassistant.companion.android.util.homeAssistantInstance2
 
 @Composable
 fun DiscoveryView(
     onboardingViewModel: OnboardingViewModel,
     manualSetupClicked: () -> Unit,
     instanceClicked: (instance: HomeAssistantInstance) -> Unit
+) {
+    DiscoveryView(foundInstances = onboardingViewModel.foundInstances, manualSetupClicked = manualSetupClicked, instanceClicked = instanceClicked)
+}
```
[the remark this answers is not in the document]
Comment (vieck) [reply app/src/main/java/io/homeassistant/companion/android/onboarding/discovery/DiscoveryView.kt:46]: @jpelgrom This is only for compose previews. Basically, if we use the second one we need to expand the parameters of un DiscoveryView(
    onboardingViewModel: OnboardingViewModel,
    manualSetupClicked: () -> Unit,
    instanceClicked: (instance: HomeAssistantInstance) -> Unit
) {
to include everything that we use in onboardingViewModel

[the remark this answers is not in the document]
Comment (vieck) [reply app/src/main/java/io/homeassistant/companion/android/onboarding/discovery/DiscoveryView.kt:46]: @jpelgrom Read here real quick https://developer.android.com/jetpack/compose/tooling/previews#preview-viewmodel
"If you want to preview a composable that uses a ViewModel, you should create another composable with the parameters from ViewModel passed as arguments of the composable. This way, you don't need to preview the composable that uses the ViewModel."

File: app/src/main/java/io/homeassistant/companion/android/onboarding/manual/ManualSetupView.kt [hunk 5]
```diff
@@ -30,6 +35,16 @@ import io.homeassistant.companion.android.onboarding.OnboardingViewModel
 fun ManualSetupView(
     onboardingViewModel: OnboardingViewModel,
     connectedClicked: () -> Unit
+) {
+    ManualSetupView(onboardingViewModel.manualUrl, onboardingViewModel::onManualUrlUpdated, onboardingViewModel.manualContinueEnabled, connectedClicked)
+}
```
[the remark this answers is not in the document]
Comment (vieck) [reply app/src/main/java/io/homeassistant/companion/android/onboarding/manual/ManualSetupView.kt:40]: See https://github.com/home-assistant/android/pull/4239#discussion_r1519010004

File: app/src/main/java/io/homeassistant/companion/android/onboarding/discovery/DiscoveryView.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (jpelgrom) [reply app/src/main/java/io/homeassistant/companion/android/onboarding/discovery/DiscoveryView.kt:46]: Please update this as requested (or give more arguments to convince why not :)) to proceed.

**Review by jpelgrom — CHANGES_REQUESTED** [review]

**Review by jpelgrom — APPROVED** [review]

**Pull request merged**